### DISCUSSION TITLE
New attribution requirement and MIT license conflicts

### DISCUSSION THREAD

**Discussion by jamieshepherd** [xyflow/xyflow#2015] [Q&A] [answered]
Hi,

First of all, I respect that the library is open source, but I want to be clear on the licensing aspect as it currently stands. The public repo is released under the MIT license, which grants permission `free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software`.

On the react flow pricing page, there is a section reading:

```
We expect users who are using React Flow commercially to subscribe to React Flow Pro if they want to remove the attribution. In non-commercial applications you may hide the attribution without subscribing but are welcome to [sponsor us on Github](https://github.com/sponsors/wbkd).
```

Legally speaking, this seems like an "ask" rather than a requirement, as the MIT license would at least allow any user to download and modify at will, regardless of commercial status. I just want to solidify the actual requirements here.

Disclosure: Not using react-flow commercially.

**Comment by moklick** [comment, marked as answer]
Hey @jamieshepherd 

thanks for your feedback! React Flow Pro is only a few days old and we are open to revise our ideas. After working on this library voluntarily for two years, we are now looking for ways to finance the further maintenance and development. Companies that use React Flow now have the chance to decide wether they want to use a side project or a professionally maintained library. We won't change the MIT license so you can do whatever you want with it. Why is your conclusion not to use it commercially?

**Reply by jamieshepherd** [reply]
Thanks for your reply, I absolutely understand the need to finance development - open source is a hard and often unforgiving time. The only reason for this post was to clear up some confusion between the two conflicting statements - to alleviate any concerns I or others might have for using the library commercially in the future.

> Why is your conclusion not to use it commercially?

This was a disclosure, not a conclusion. I'm letting you know that I'm not currently using react-flow commercially (but could, in the future).

Thanks again, I've marked your reply as answered.

**Reply by moklick** [reply]
Got it! Thanks for the explanation. It's an experiment :) we will see how it goes.

**Comment by pdparchitect** [comment]
Team, congrats on the launch.

I have the same concerns as the OP. We need more clarity where you stand. While from licensing perspective it seems that forking is an option, it is certainly not something anyone who respects your work will ever do. In my opinion you either need to change the license or find another way to introduce your commercial options.

We also need clarity on how you are planning to track unlicensed use of your library. Are you going to include telemetry or are you planning to rely on the good will of people. Including any tracking, especially if not fully documented, will not go well with anyone.

My last feedback is that your licensing option is very much enterprise oriented and in my opinion ultimately prevents your library to find a foothold in potentially very successful products where paying for the library will be 100% guaranteed as long as these products are ultimately successful. Perhaps you can look into providing licensing mechanism similar to what Unreal currently use where fees are due only if used in successful commercial settings above certain revenue threshold.

**Reply by chrtze** [reply]
Hi @pdparchitect,

thanks for your feedback. We are currently working on improvements for our subscription model and will update the features soon. To clarify upfront: React Flow is open source and MIT-licensed. The only thing that we are demanding at the moment is to only hide the attribution if you are subscribed. See @moklick's answer for more details.

> While from licensing perspective it seems that forking is an option, it is certainly not something anyone who respects your work will ever do.

That is exactly what we want to achieve. Keeping the library MIT Licensed while trying to get companies who are making money with React Flow to finance the development.

> In my opinion you either need to change the license or find another way to introduce your commercial options.

We currently do not want to change the license as it would make things way more complicated for all users of React Flow. We are already working on providing more value for our subscribers.

> We also need clarity on how you are planning to track unlicensed use of your library. Are you going to include telemetry or are you planning to rely on the good will of people. Including any tracking, especially if not fully documented, will not go well with anyone.

We are not planning on implementing tracking or anything like it. We have decided to introduce a subscription model to give users of React Flow the certainty that the library will be developed further in the future.

> My last feedback is that your licensing option is very much enterprise oriented and in my opinion ultimately prevents your library to find a foothold in potentially very successful products where paying for the library will be 100% guaranteed as long as these products are ultimately successful.

We are not selling licenses for React Flow. It is published under MIT license so that everyone (including enterprises) can easily use it in their codebase. For early startups, we are offering a discount as well. If you don't want to subscribe, you can use React Flow with attribution for free.

I am wondering how we can improve our documentation to make sure that there is less confusion like this. I know that this is a bit of an unusual business model but worth a try. Any thoughts?

**Reply by Tenkir** [reply]
> The only thing that we are demanding at the moment is to only hide the attribution if you are subscribed.

This is against the idea of an MIT license. With an raw MIT license, you are welcome to **request** that users do not remove the attribution, but that cannot be a requirement of the license.

If that is the desire of the maintainers, perhaps the license should be changed to something which would allow these types of restriction.

----

For the record, I am not against a paid tier (we sponsor it at Little Cinema Digital), it's just a bit of a surprise when updating a library which was MIT licensed, and now having new restrictions placed on the library.

**Reply by moklick** [reply]
Hey @Tenkir 

we are aware that the wording about the attribution is not perfect currently. We are already working on this https://github.com/wbkd/react-flow/issues/2288 

Thanks for your sponsorship!

**Reply by smadapathi** [reply]
Hi @moklick, can we remove the attribution for our commercial project, without pro subscription?

**Comment by lgj9172** [comment]
I also have a same question.
I heard that anybody can change a code in MIT license.
But in FAQ, I should use pro plan if i would remove attribution tag.
Actually, I can fork this project and revise to not show attribution tag anymore because it is MIT license.
Can I do this?
I think it is conflict point.

**Reply by moklick** [reply]
Hey! We will adjust the wording with the next update. You don't need to do a subscription if you want to remove the attribution. However if you are using ReactFlow commercially, we would appreciate a subscription or github sponsorship, so that we can ensure the further development.

**Reply by smadapathi** [reply]
Thanks for your reply @moklick. As of now, we would like to keep the attribution and thanks for this library again :)
